htdocs/core/class/commoninvoice.class.php

```diff
@@ -637,6 +638,84 @@ public function getLibType($withbadge = 0)
 		return $out;
 	}
 
+	/**
+	 *	Return label of invoice subtype
+	 *
+	 *  @param		string		$table          table of invoice
+	 *	@return     string        				Label of invoice subtype
+	 */
+	public function getSubtypeLabel($table = '')
+	{
+		if ($table === 'facture' || $table === 'facture_fourn') {
+			$sql = "SELECT s.label FROM " . MAIN_DB_PREFIX . $table . " AS f";
+			$sql .= " INNER JOIN " . MAIN_DB_PREFIX . "c_invoice_subtype AS s ON f.subtype = s.rowid";
+			$sql .= " WHERE f.ref = \"$this->ref\"";
```

Comment:
Possible SQL Injection here.
Use
`$sql .= " WHERE f.ref = '".$this->db->escape($this->ref)."'";`

Reply:
Fixed